src/main/java/seedu/duke/Manager.java

```diff
@@ -138,6 +138,17 @@ private Status executeCommand(String commandWord, String parameters) throws Halp
             command = Parser.parseFindMedicine(parameters);
             status = command.execute(storage.medicines);
             break;
+        case "dispense medicine":
+            command = Parser.parseDispenseMedicine(parameters);
+
+            Command helperCommand = Parser.parseCheckForPatientAppointment(parameters);
+            helperCommand.execute(storage.appointments);
+            helperCommand = Parser.parseCheckMedicineStock(parameters);
+            helperCommand.execute(storage.medicines);
+
+            command.execute(storage.appointments);
+            status = command.execute(storage.medicines);
```

Comment:
Maybe extract this as a method if it's possible? so that our case switch isn't as long

Reply:
I think this can be done in another PR and account for the other commands as well.